src/wakefs/squashfuse_helper.cpp

```diff
@@ -0,0 +1,67 @@
+#define _XOPEN_SOURCE 700
+#define _POSIX_C_SOURCE 200809L
+
+#include "squashfuse_helper.h"
+
+#include <errno.h>
+#include <fcntl.h>
+#include <libgen.h>
+#include <poll.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <wcl/defer.h>
+#include <wcl/result.h>
+#include <wcl/xoshiro_256.h>
+
+#include <iostream>
+
+void cleanup_fifo(std::string squashfuse_fifo_path, int squashfuse_notify_pipe_fd) {
+  close(squashfuse_notify_pipe_fd);
+  unlink(squashfuse_fifo_path.c_str());
+}
+
+// Works like mkfifo but it gets created in a mktemp() style.
+wcl::result<std::string, wcl::posix_error_t> mktempfifo() {
+  wcl::xoshiro_256 rng(wcl::xoshiro_256::get_rng_seed());
+  std::string fifo_filepath = "/tmp/squashfuse_notify_pipe_fifo_" + rng.unique_name();
+
+  int mkfifoat_result = mkfifo(fifo_filepath.c_str(), 0664);
+  if (mkfifoat_result < 0) {
+    return wcl::make_errno<std::string>();
+  }
+
+  return wcl::make_result<std::string, wcl::posix_error_t>(fifo_filepath);
+}
+
+wcl::optional<SquashFuseMountWaitError> wait_for_squashfuse_mount(
+    const std::string& squashfuse_fifo_path) {
+  int squashfuse_notify_pipe_fd = open(squashfuse_fifo_path.c_str(), O_RDONLY);
+  if (squashfuse_notify_pipe_fd == -1) {
+    return wcl::some(SquashFuseMountWaitError{SquashFuseMountWaitErrorType::CannotOpenFifo, errno});
+  }
```

Comment:
You can use wcl::unique_fd instead of open and then you can just unlink in `make_defer`

Reply:
Made the changes